securedrop_client/api_jobs/downloads.py

```diff
@@ -40,6 +41,24 @@ def __init__(self, data_dir: str) -> None:
         super().__init__()
         self.data_dir = data_dir
 
+    def _get_realistic_timeout(self, size_in_bytes: int) -> int:
+        """
+        Returns a realistic timeout in seconds for a file, message, or reply based on its size.
+
+        Note that:
+        * the size of the file provided by server is in bytes (the server computes it using
+        os.stat.ST_SIZE)
+        * A reasonable time to download a 1 MB file download time according to OnionPerf was
+        ~7.5s:
+        https://metrics.torproject.org/torperf.html?start=2019-05-02&end=2019-07-31&server=onion&filesize=1mb
+
+        This simply scales the timeouts per file, so that a small file times out much faster than a
+        large file, which is given a long time to complete.
+        """
+
+        SCALING_FACTOR_FILE_DL_TIME_SECS_PER_MB = 7.5
```

Comment:
This PR looks great! Would just like to discuss how you landed at 7.5 (and 5 for api_client.default_request_timeout)

Reply:
so these estimated figures are eyeballed from OnionPerf, which is run by Tor to track performance of the Tor network + onion services. The 7.5 seconds per 1 MB scaling factor comes from recent [OnionPerf measurements](https://metrics.torproject.org/torperf.html?start=2019-05-02&end=2019-07-31&server=onion&filesize=1mb) (our scaling factor shown in dotted line):

<img width="969" alt="Screen Shot 2019-08-05 at 5 14 12 PM" src="https://user-images.githubusercontent.com/7832803/62495746-deae5f80-b7a4-11e9-82dc-90d5ddb9627b.png">

And the (admittedly generous since not all these requests/responses go through the queue right now thus they don't all get the automatic retry behavior upon timeout) 5 second number for generic requests/responses comes from smaller request measurements to onion servers:

<img width="935" alt="Screen Shot 2019-08-05 at 5 18 29 PM" src="https://user-images.githubusercontent.com/7832803/62495828-11585800-b7a5-11e9-99f2-07c049991370.png">